### ISSUE TITLE
oohttp: add header ordering support

### ISSUE THREAD

**Issue by fedosgad** [ooni/probe#2160]
I have forked [oohttp](https://github.com/ooni/oohttp) (this is a great library btw, thanks a lot for it!) to add support for features absent in stdlib's http client, such as header ordering (including HTTP2 headers and pseudoheaders), manipulating HTTP2 settings and HTTP2 SERVER PUSH. My intent is to make mimicking "normal" web browsers (Chrome/FF/you name it) simplier and more accurate.

Code is available at [github.com/fedosgad/oohttp](https://github.com/fedosgad/oohttp). Currently, I've finished header ordering (ported from [fhttp](https://github.com/useflyent/fhttp)).

Changes were made in (I hope) maintainable way, similar to what oohttp itself has done (see [PATCHING.md](https://github.com/fedosgad/oohttp/blob/main/PATCHING.md)). I would like to contribute those changes back to oohttp.  What do you think about it?

**Comment by bassosimone** [comment]
Absolutely! I'll add to my sprint to take a look at your changes, thank you!

**Comment by fedosgad** [comment]
That's great! Feel free to ping me if there are any questions.

**Comment by nathanejohnson** [comment]
I am very much interested in seeing this get rolled in.  @bassosimone would it be helpful for me to put together a PR, or is this something you've already started on?

**Comment by bassosimone** [comment]
@nathanejohnson it seems there is already an open PR: https://github.com/ooni/oohttp/pull/46

**Comment by hellais** [comment]
Unfortunately we are going to be discontinuing our oohttp golang std lib http fork, as it's too expensive for us to maintain.

As a result I am closing this issue.

**Comment by fedosgad** [comment]
Sad to hear this but thanks for the early warning!
Do you mind sharing what libraries you will use instead of oohttp?
